lib/gems/pending/util/object_storage/miq_swift_storage.rb

```diff
@@ -0,0 +1,148 @@
+require 'pp'
+# require 'util/miq_object_storage'
+require 'util/mount/miq_generic_mount_session'
+
+class MiqSwiftStorage < MiqObjectStorage
+  attr_reader :container_name
+
+  def self.uri_scheme
+    "swift".freeze
+  end
+
+  def self.new_with_opts(opts)
+    new(opts.slice(:uri, :username, :password))
+  end
+
+  def initialize(settings)
+    super(settings)
+
+    # NOTE: This line to be removed once manageiq-ui-class region change implemented.
+    @bucket_name         = URI(@settings[:uri]).host
+
+    raise "username and password are required values!" if @settings[:username].nil? || @settings[:password].nil?
+    _scheme, _userinfo, @host, @port, _registry, @mount_path, _opaque, query, _fragment = URI.split(URI.encode(@settings[:uri]))
+    query_params(query)
+    @swift          = nil
+    @username       = @settings[:username]
+    @password       = @settings[:password]
+    @container_name = @mount_path[0] == File::Separator ? @mount_path[1..-1] : @mount_path
+  end
+
+  def uri_to_local_path(remote_file)
+    # Strip off the leading "swift:/" from the URI"
+    File.join(@mnt_point, URI(remote_file).host, URI(remote_file).path)
+  end
+
+  def uri_to_object_path(remote_file)
+    # Strip off the leading "swift://" and the container name from the URI"
+    # Also remove teh leading delimiter.
+    object_file_with_bucket = URI.split(URI.encode(remote_file))[5]
+    object_file_with_bucket.split(File::Separator)[2..-1].join(File::Separator)
+  end
+
+  def upload_single(dest_uri)
+    #
+    # Get the remote path, and parse out the bucket name.
+    #
+    object_file = uri_to_object_path(dest_uri)
+    #
+    # write dump file to swift
+    #
+    logger.debug("Writing [#{source_input}] to => Bucket [#{container_name}] using object file name [#{object_file}]")
+    begin
+      swift_file = container.files.new(:key => object_file)
+      params     = {
+        :expects       => [201, 202],
+        :headers       => {},
+        :request_block => -> { read_single_chunk },
+        :idempotent    => false,
+        :method        => "PUT",
+        :path          => "#{Fog::OpenStack.escape(swift_file.directory.key)}/#{Fog::OpenStack.escape(swift_file.key)}"
+      }
+      swift_file.service.send(:request, params)
+      clear_split_vars
+    rescue Excon::Errors::Unauthorized => err
+      logger.error("Access to Swift container #{@container_name} failed due to a bad username or password. #{err}")
+      msg = "Access to Swift container #{@container_name} failed due to a bad username or password. #{err}"
+      raise err, msg, err.backtrace
+    rescue => err
+      logger.error("Error uploading #{source_input} to Swift container #{@container_name}. #{err}")
+      msg = "Error uploading #{source_input} to Swift container #{@container_name}. #{err}"
```

Comment:
Couldn't we deduplicate the error message here by assigning `msg` first?:
```ruby
msg = "Error uploading #{source_input} to Swift container #{@container_name}. #{err}"
logger.error(msg)
raise err, msg, err.backtrace
```

Reply:
Looks like this applies to quite a few places, please fix all of them.

Reply:
Fixed in six places.